### ISSUE TITLE
False-positive: "objects are always passed as reference"

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#462]
It should consider when you wants to fill a new variable with this reference.
I think that you could fix it easily by checking if the parameter variable is rewrited on scope context.

For instance:

```php
function now(Carbon &$carbon) {
    $carbon = Carbon::now(); // <-- This should disable the $carbon inspection for that.
}

// $now is undefined here:
now($now);
```

**Comment by kalessil** [comment]
Great catch @rentalhost, thanks!
